src/cpp/src/lm_encoding.cpp

```diff
@@ -203,11 +203,19 @@ std::pair<EncodedResults, int32_t> get_lm_encoded_results(
         raw_perf_counters.m_batch_sizes.emplace_back(batch_size);
 
         if (streamer_ptr) {
-            // stream data from first sequence
-            int64_t out_token = sequence_groups.at(0).get()->operator[](0)->get_generated_ids().back();
-            if (streamer_ptr->put(out_token)) {
-                break;
+            // not generated tokens like several prompt phase
+            if (!generations.at(0).get()->can_read()) {
+                continue;
             }
+            std::unordered_map<uint64_t, GenerationOutput> token = generations.at(0).get()->back();
+            OPENVINO_ASSERT(1 <= token.size());
+            OPENVINO_ASSERT(1 <= token.begin()->second.generated_ids.size());
+            for (const auto& gen_token : token.begin()->second.generated_ids) {
+                if (!streamer_ptr->put(gen_token)) {
+                    break;
+                }
+            }
+            
```

Comment:
can we also update CB impl
https://github.com/openvinotoolkit/openvino.genai/blob/d17f7168f278ef98acfdc7ba1ac93e4c759a6402/src/cpp/src/continuous_batching_impl.cpp#L279-L292
as it seems to have the same issue - assumes that only 1 tokens can be generated at once